pkg/querier/queryrange/split_by_interval_test.go

```diff
@@ -269,6 +269,68 @@ func Test_splitMetricQuery(t *testing.T) {
 	}
 }
 
+func Benchmark_splitByInterval_Do(b *testing.B) {
```

Comment:
It would be great to have benchmarks for other cases as well, not just merging on empty response with one non-empty response. I'd suggest:
`2 empty`
`1 empty/1 non empty`
`2 non empty`